ovs/lib/setup.py

```diff
@@ -1436,10 +1659,27 @@ def _configure_logstash(client):
         Toolbox.change_service_state(client, 'logstash', 'restart', logger)
 
     @staticmethod
+    def _unconfigure_logstash(client):
+        if not SetupController._logstash_installed(client):
+            logger.debug("Logstash is not installed, skipping its configuration")
+            return False
+
+        print 'Unconfiguring logstash'
+        logger.info('Unconfiguring logstash')
+        Toolbox.change_service_state(client, 'logstash', 'stop', logger)
+        if ServiceManager.has_service('logstash', client) is True:
+            ServiceManager.remove_service('logstash', client)
+
+
+    @staticmethod
     def _configure_avahi(client, cluster_name, node_name, node_type):
         print '\n+++ Announcing service +++\n'
         logger.info('Announcing service')
 
+        if client.file_exists(SetupController.avahi_filename):
```

Comment:
This method is also used for reconfiguring Avahi. So if it was already configured as ```master```, and it's called to reconfigure it as ```extra```, it will skip this, right?